### PR TITLE
Get node status from tenant cluster api

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/6893

### CODE REVIEW

File: service/health/searcher/types.go [hunk 1]
```diff
@@ -47,3 +50,24 @@ func NewClusterStatus(cluster clusterInfo) ClusterStatus {
 		NodeCount: currentNodes,
 	}
 }
+
+func NewNodeStatus(node v1.Node) NodeStatus {
+	ready := false
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady {
+			ready = condition.Status == "True"
```
Comment (stone-z) [inline service/health/searcher/types.go]: Double-checking this logic - is it the case that any "Ready" condition in this list makes the node ready, despite other conditions?

Comment (tfussell) [reply service/health/searcher/types.go]: Based on https://kubernetes.io/docs/concepts/architecture/nodes/#condition , I think this is the case. I haven't experienced enough unhealthy nodes to say for sure. Do you know @MarcelMue?

File: service/health/searcher/service.go [hunk 2]
```diff
@@ -121,8 +133,25 @@ func (s *Service) Search(ctx context.Context, request Request) (*Response, error
...
+			return nil, microerror.Mask(err)
+		}
+		nodes, err := k8sClient.CoreV1().Nodes().List(v1.ListOptions{})
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		cluster.nodes = nodes.Items
+	}
+
+	fmt.Println(cluster.nodes)
```
Comment (MarcelMue) [inline service/health/searcher/service.go]: Remove when done.

File: service/health/searcher/service.go [hunk 3]
```diff
@@ -121,8 +133,25 @@ func (s *Service) Search(ctx context.Context, request Request) (*Response, error
 		}
 	}
 
+	{
+		// TODO: cache this?
```
Comment (MarcelMue) [inline service/health/searcher/service.go]: Maybe later, not necessary now.

File: service/health/searcher/service.go [hunk 4]
```diff
@@ -121,8 +133,25 @@ func (s *Service) Search(ctx context.Context, request Request) (*Response, error
...
+		var k8sClient kubernetes.Interface
```
Comment (MarcelMue) [inline service/health/searcher/service.go]: Why var? You assign right in the next line.

File: service/health/searcher/types.go (hunk 1 above)
Comment (MarcelMue) [reply service/health/searcher/types.go]: This is explicitly checking for the `Ready` condition on the node - which defines readiness. It is not checking for all conditions in the loop. Therefor this should be fine.

**Review by stone-z — APPROVED** [review]

**Review by MarcelMue — APPROVED** [review]
Okay for now, we iterate.

**Pull request merged**